### PR TITLE
Ethereum Oracle protocol

### PR DESCRIPTION
**Pull request by Abhishekkochar**
Ethereum data will be fetched. More time is needed for the reward mechanism in the polygon.

### CODE REVIEW

File: oracle/collector/ethereum/index.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const {collect_uniSwap_aave_crv} = require('./uniSwap')
+
+async function collect_eth{
```
Comment (goeth121) [inline oracle/collector/ethereum/index.js:3]: this kind of function implementation works? I guess `collect_eth() {}` should be working

Comment (gophereth) [inline oracle/collector/ethereum/index.js:3]: Probably ` collect_eth{` to ` collect_eth() {`?

File: oracle/collector/ethereum/uniSwap/truEth.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        const ethDecimals = await ethContract.methods.decimals().call()
+        const truDecimals = await truContract.methods.decimals().call()
+        const reserves = await poolContract.methods.getReserves().call()
+        const totalEth = await reserves[0]/Math.pow(10, ethDecimals)
+        const totalTru = await reserves[1]/Math.pow(10, truDecimals)
+        //calculating total liquidity
+        const totalLiquidity = totalEth*ethPrice + totalTru* truPrice
+        console.log('Total liquidty of the pool:  ' + totalLiquidity)
+        const lpTokenPrice = totalLiquidity/totalSupply
+        console.log('LP token price: ' + lpTokenPrice)
```
Comment (gophereth) [inline oracle/collector/ethereum/uniSwap/truEth.js:51]: TRU and SUSHI reward rate should be calculated

File: oracle/collector/polygon/quickSwap/ethUsdc.js [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+//address
+//const mainFarmAddress = '0x5757371414417b8C6CAad45bAeF941aBc7d3Ab32'
+const contractAddress = '0x853Ee4b2A13f8a742d64C8F088bE7bA2131f670d'
+//const usdcAddress = '0x2791Bca1f2de4661ED88A30C99A7a9449Aa84174'
+const ethAddress = '0x7ceB23fD6bC0adD59E62ac25578270cFf1b9f619'
+const usdcPriceFeedAddress = '0xfE4A8cc5b5B2366C1B58Bea3858e81843581b2F7'
+const ethPriceFeedAddress = '0xF9680D99D6C9589e2a93a78A04A279e509205945'
+const rewardAddress = '0xbB703E95348424FF9e94fbE4FB524f6d280331B8'
+
+async function quickSwap_eth_usdc_collector(){
```
Comment (gophereth) [inline oracle/collector/polygon/quickSwap/ethUsdc.js:20]: Can we change the function name to `quickswap_eth_usdc_collector`

File: oracle/collector/ethereum/uniSwap/truEth.js (hunk 2 above)
Comment (Abhishekkochar) [reply oracle/collector/ethereum/uniSwap/truEth.js:51]: okay, that should be alright. I will let you know once it's done.

**Comment by Abhishekkochar** [comment]
Hi @gophereth, 

Please review these changes.

Cheers,
Abhishek

**Comment by gophereth** [comment]
> Please review these changes.
> 
> Cheers, Abhishek

Thanks I will check it out shortly

**Review by gophereth — APPROVED** [review]

**Pull request merged**